src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -467,10 +469,14 @@ impl Coordinator {
         let ingestion_id = source.global_id();
         let subsource_stmts = generate_subsource_statements(&scx, source_name, subsources)?;
 
+        let id_ts = self.get_catalog_write_ts().await;
+        let mut ids = self
+            .catalog_mut()
+            .allocate_user_ids(u64::cast_from(subsource_stmts.len()), id_ts)
+            .await?;
         for subsource_stmt in subsource_stmts {
-            let s = self
-                .plan_subsource(session, &params, subsource_stmt)
-                .await?;
+            let (item_id, global_id) = ids.pop().expect("allocated sufficient ids");
+            let s = self.plan_subsource(session, &params, subsource_stmt, item_id, global_id)?;
             subsource_plans.push(s);
         }
```

Comment:
to retain the ID ordering we had before you could zip `ids` and `subsource` statements together, would eliminate the need for the `.pop`

Reply:
Yup, zipping is the obvious choice here!